.github/workflows/upstream-sync.yml

```diff
@@ -3,86 +3,86 @@ name: Upstream Sync Workflow
-                owner,
-                repo,
-                head: '${{ steps.branch-name.outputs.result }}',
-                base: '${{ format('{0}/{1}', env.downstream-prefix, inputs.release-series) }}',
-                maintainer_can_modify: false,
-                body: [
-                  'The changes from ${{ env.upstream-prefix }}/${{ inputs.release-series }} have been included in this automated PR'
-                ].join('\n')
-              });
-              github.rest.issues.addLabels({
-                owner,
-                repo,
-                issue_number: result.data.number,
-                labels: ['automated', 'sync']
-              });
-            }
+        id: get_upstream_sha
+        run: |
+          upstream_sha=$(gh api \
+          -H "Accept: application/vnd.github.v3+json" \
+          "/repos/${{env.UPSTREAM_OWNER}}/$(basename $(pwd))/commits/stable/${{inputs.release_series}}" --jq ".sha")
+          echo "::set-output name=result::$upstream_sha"
+      - name: Check if downstream branch exists
+        id: check_if_downstream_branch_exists
+        run: |
+          api_path="/repos/${{env.DOWNSTREAM_OWNER}}/$(basename $(pwd))/branches/stackhpc/${{inputs.release_series}}"
+          response=$(gh api -H "Accept: application/vnd.github.v3+json" $api_path --silent --include) || true
+          if [[ "$response" =~ "HTTP/2.0 200 OK" ]]; then
+            echo "::set-output name=result::true"
+          elif [[ "$response" =~ "HTTP/2.0 404 Not Found" ]]; then
+            echo "::set-output name=result::false"
+          else
+            exit 1
+          fi
+      - name: Create missing downstream branch
+        if: steps.check_if_downstream_branch_exists.outputs.result == 'false'
+        run: |
+          gh api \
+          --method POST \
+          -H "Accept: application/vnd.github.v3+json" \
+          "/repos/${{env.DOWNSTREAM_OWNER}}/$(basename $(pwd))/git/refs" \
+          -f ref='refs/heads/stackhpc/${{inputs.release_series}}' \
+          -f sha=${{steps.get_upstream_sha.outputs.result}}
+      - name: Check if upstream is ahead of downstream
+        if: steps.check_if_downstream_branch_exists.outputs.result == 'true'
+        id: check_if_ahead
+        run: |
+          ahead_by=$(gh api \
+          -H "Accept: application/vnd.github.v3+json" \
+          "/repos/${{env.DOWNSTREAM_OWNER}}/$(basename $(pwd))"\
+          "/compare/stackhpc/${{inputs.release_series}}..."\
+          "${{env.UPSTREAM_OWNER}}:stable/${{inputs.release_series}}" --jq '.ahead_by')
+          echo "::set-output name=result::$ahead_by"
+      - name: Create copy of the upstream branch
+        if: steps.check_if_ahead.outputs.result > 0
+        run: |
+          gh api \
+          --method POST \
+          -H "Accept: application/vnd.github.v3+json" \
+          "/repos/${{env.DOWNSTREAM_OWNER}}/$(basename $(pwd))/git/refs" \
+          -f ref="refs/heads/stackhpc/${{inputs.release_series}}-$(date +%F)" \
```

Comment:
If this branch is called `stackhpc/yoga-2022-05-23`, won't it match the protected branch name pattern?

Reply:
Either is fine